### DISCUSSION TITLE
Add actions to Markdown card

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/feature-requests#713] [Dashboards & cards]
using Markdown cards as header cards, for their ability of templates and dynamic titles.

would be magic if we could hold or tap the header to navigate away

and be able to do:
```
  - type: markdown
    card_mod: !include /config/dashboard/card_mods/markdown_heading.yaml # add some styling
    content: In 1 oogopslag
    tap_action:
      action: navigate
      navigation_path: /ui-data/samenvatting
    hold_action:
      action: navigate
      navigation_path: /ui-sub-views/samenvatting_groepen
```

as we now have to resort to `custom:button-card` to do the same. The new section heading card is no alternative either, because we can not use dynamic content.

Ofc that would also be magic ;-)

**Comment by dsellers1** [comment]
I've thought the same Marius. The Markdown card is a really useful and powerful card, but, when I want to use an action I, too, find myself resorting the the custom:button-card. 

I would add to the request that the default actions should be set to none.

**Comment by BBE-FR** [comment]


**Reply by Mariusthvdb** [reply]
please dont mix FR's

**Reply by BBE-FR** [reply]
Did not see the issue to link another FR for context. Will not interfere, and thus remove the link...

But for my understanding,  
Your exact word were: "using Markdown cards as header cards, for their ability of templates and dynamic titles."
Why linking an FR related to "templating into header card" is an issue? 
Both FR reflect the same need, with two différents approches. The first one ready could cover your need...

**Reply by Mariusthvdb** [reply]
Sorry, didn’t mean to be too strict. It’s just that adding multiple FR in a single thread can be hard to follow for the devs.

keep it small and simple is the recipe for success in these cases

(Btw templates will probably never be introduced in these cards as it’s considered too complex and advanced for day to day use and maintainability) I did +1 though ;-)

**Comment by BBE-FR** [comment]
As a workaround, instead of button card, some user could use some mushroom card to allow templating and "tap-action customisation.
These cards do not offer exactly the same capabilities than the button card, but allows full graphical interface support (for YAML reluctant people)

Two alternatives for this useage
- Mushroom title card (that support templating into the title, the subtitle, and tap action on both title and subtitle) for a header style:
https://github.com/piitaya/lovelace-mushroom/blob/main/docs/cards/title.md

- mushroom template card, For a more "standard" card look, with borders and background, with an additional icon and badge to be customized:
https://github.com/piitaya/lovelace-mushroom/blob/main/docs/cards/template.md

**Reply by Mariusthvdb** [reply]
> As a workaround,

dont think we should list all possible custom options here.
This is a FR to the core Markdown card, and the request is plain and simple: add [the actions options](https://www.home-assistant.io/dashboards/actions/) to the card.

Now if we could maybe join and open a PR for that, I suppose we would have a start.
I had a few Frontend PR's merged myself, but they were very simple, and nothing compared to adding this card functionality.

Derek, would you know how to proceed doing that?

[Glance card](https://github.com/home-assistant/frontend/blob/dev/src/panels/lovelace/cards/hui-glance-card.ts) has a lot on actions...

**Reply by mirekmal** [reply]
I found this threead searching for solution to my problem. I'm using markdown card to build table of movies in my library. So far I succeeded to create nice table with posters, but element that I'm missing is clickable action. For example one of the columns in table contains the link to IMDB (easy, as it is link to external server and I can template easily movi ID into the link), but I'd like to be able to click on movie poster (or other element, could be text) and be redirected to another dashboard subview with details of specific movie. This require to embed the mnovie ID into link and HA cannot handle this as it would require to have dashboard page created for every movie (>6k in my case). Service call which allows to pass to service which element was tapped would solve the problem. So my idea goes step further as not entire card would be clickable, but each of elements in markdown could be as such.
To be honest I tried to overcome this by creating a templatable link that would be fake webhook to HA and then webhook trigered automation could handle the rest. But I faled with creation of working wedhook URL to be inserted to markdown. Perhaps this would be solution? If possible...

**Reply by Mariusthvdb** [reply]
yep, as FR: add actions to Markdown card.
If that would be added, you could do anything (well, except for templates, for that you need to go to custom:button-card probably)

**Reply by Mariusthvdb** [reply]
guess we need to move to the new discussion section in the home assistant repo

**Reply by mirekmal** [reply]
FR already created ~2 months ago...
https://github.com/orgs/home-assistant/discussions/47

**Reply by Mariusthvdb** [reply]
I now see I added my comment to your post, and not my main FR here. Sorry for the mixup, Ill move it so it doesnt get burried in this side chat

**Comment by Mariusthvdb** [comment]
Move to the new discussion section in the home assistant repo

https://github.com/orgs/home-assistant/discussions/692

**Reply by karwosts** [reply]
If you prefer we can migrate an existing request instead of duplicating it. 

I haven't seen any guidance if one forum is preferred over the other though.

**Reply by Mariusthvdb** [reply]
> If you prefer we can migrate an existing request instead of duplicating it.

o yes please, that is what we've asked but were told not possible...

there is a[ community post on the migration of FR's](https://community.home-assistant.io/t/heads-up-feature-requests-are-moving/903057), which should no longer be done in the Community section (which will be closed as is, no migration whatsoever), or in here Dashboard discussion for that matter.

New ones should go in HomeAssistant discussion in GithHub repo.

Hence I followed suit for this one.

**Reply by karwosts** [reply]
There's no migration from the forums, but github to github is just a click.

I tried it for this for an experiment. Looks like it reset the votes to 1 though 😢

**Comment by Mariusthvdb** [comment]
~havent tried it yet, but maybe~ we can use this custom card (I know, yet another custom resource...)
https://github.com/nutteloost/actions-card

```
type: custom:actions-card
card:
  type: markdown
  content: >-
    The **Markdown** card allows you to write any text. You can style it
    **bold**, *italicized*, ~strikethrough~ etc. You can do images, links, and
    more.


    For more information see the [Markdown
    Cheatsheet](https://commonmark.org/help).
hold_action:
  action: navigate
  navigation_path: /ui-overzicht/welkom
```
works as hoped for

**Comment by ildar170975** [comment]
https://github.com/home-assistant/frontend/pull/28951
